Automattic/studio
# Add Sync Push Site

## Related issues

- Fixes https://github.com/Automattic/dotcom-forge/issues/9356

## Proposed Changes

- Add push state
- Add polling for restoring backup progress
- Display Progress, Success and Error messages
- Refactor states to re-use pull logic
- Blocks import/export, disconnect and other pull/push actions. I'm not blocking start/stop.
- Display a notification after the successful push


## Testing Instructions



You can try with a site up to 2GB in size.

- Run `STUDIO_SITE_SYNC=true npm start`
- Click on the Sync tab
- Connect a site
- Click on Push
- Change tabs and sites, observe the process continues correctly
- Wait until the process finishes
- Observe your remo site has restored. In fact, the remote site would look the same as before, but it restored a previous backup of the same site.

https://github.com/user-attachments/assets/ce3479de-a0fd-4bb8-97de-f8b8535f376f

## Pre-merge Checklist
